semgrep-core/src/core/Semgrep_core_response_util.ml

```diff
@@ -0,0 +1,74 @@
+(*
+   Utilities for working with the types defined in Semgrep_core_response.atd
+   (Semgrep_core_response_t module)
+*)
+
+open Semgrep_core_response_t
+
+let compare_position (a : position) b = Int.compare a.offset b.offset
+
+let compare_location (a : location) b =
```

Comment:
can't we use deriving ord to automate some of this boilerplate?

Reply:
I don't think anyone added this to atdgen. I'm sure something can be done about it. I still don't know the details about ppx syntax, but it would be a matter of supporting annotations like
```ocaml
type bar = {
  foo: foo;
} <ocaml ppx="[@@deriving compare]">

type bar = [ A | B of foo ]
```
In the above, the type definitions are mutually recursive but they don't agree on the ppx annotation that the whole recursive type definition should have. It's probably no big deal, just a matter of choosing a convention. Maybe we could just require all the individual atd definitions that are merged into one recursive ocaml definition to have the same exact ppx annotations (since merging them correctly may be more difficult).

Anyway, we don't have this at the moment.

Reply:
Oh I see. Yes, it would be great to add this to atdgen, I had the need for a deriving show recently in another .atd. Could you put a TODO in a comment in the code, so we remember why we have to write this boilerplate manually.